packages/@ourworldindata/utils/src/grapherConfigUtils.test.ts

```diff
@@ -0,0 +1,348 @@
+                    map: {
+                        projection: MapProjectionName.World,
+                        hideTimeline: false,
+                    },
+                }
+            )
+        ).toEqual({ title: "Chart", map: { hideTimeline: true } })
+        expect(
+            diffGrapherConfigs(
+                {
+                    tab: GrapherTabOption.chart,
+                    map: {
+                        projection: MapProjectionName.World,
+                        hideTimeline: true,
+                    },
+                },
+                {
+                    tab: GrapherTabOption.chart,
+                    map: {
+                        projection: MapProjectionName.World,
+                        hideTimeline: true,
+                    },
+                }
+            )
+        ).toEqual({})
+    })
+
+    it("strips undefined values from the config", () => {
+        expect(
+            diffGrapherConfigs(
+                {
+                    tab: GrapherTabOption.chart,
+                    title: "Chart",
+                    subtitle: undefined,
+                },
+                { tab: GrapherTabOption.chart, title: "Reference chart" }
+            )
+        ).toEqual({ title: "Chart" })
+    })
+
+    it("doesn't diff $schema, id, version, slug or isPublished", () => {
+        expect(
+            diffGrapherConfigs(
+                {
+                    title: "Chart",
+                    $schema:
+                        "https://files.ourworldindata.org/schemas/grapher-schema.004.json",
+                    id: 20,
+                    version: 1,
+                    slug: "slug",
+                    isPublished: false,
+                },
+                {
+                    title: "Reference chart",
+                    $schema:
+                        "https://files.ourworldindata.org/schemas/grapher-schema.004.json",
+                    id: 20,
+                    version: 1,
+                    slug: "slug",
+                    isPublished: false,
+                }
+            )
+        ).toEqual({
+            title: "Chart",
+            $schema:
+                "https://files.ourworldindata.org/schemas/grapher-schema.004.json",
+            id: 20,
+            version: 1,
+            slug: "slug",
+            isPublished: false,
+        })
+    })
+
+    it("is idempotent", () => {
+        const config: GrapherInterface = {
+            tab: GrapherTabOption.chart,
+            title: "Chart",
+            subtitle: undefined,
+        }
+        const reference: GrapherInterface = {
+            tab: GrapherTabOption.chart,
+            title: "Reference chart",
+        }
+        const diffedOnce = diffGrapherConfigs(config, reference)
+        const diffedTwice = diffGrapherConfigs(diffedOnce, reference)
+        expect(diffedTwice).toEqual(diffedOnce)
+    })
+})
+
+describe("diff+merge", () => {
+    it("are consistent", () => {
```

Comment:
I was thinking that it might also be good to test some real-world examples as well (not just these dummy configs I came up with). So, maybe that could go hand in hand with property-based testing 🤝🏻